### PR TITLE
Fix #5406: Migrate keyboard hiding functionality 

### LINKED ISSUE #5406: [Feature Request]: Migrate keyboard hiding functionality
### Is your feature request related to a problem? Please describe.

SDK 33 deprecated ``InputMethodManager.SHOW_FORCED`` (see https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#SHOW_FORCED) which is used to hide the keyboard via a call to ``InputMethodManager.hideSoftInputFromWindow``.

#5402 helped reveal which activities need updating:
- [ ] app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt
- [ ] app/src/main/java/org/oppia/android/app/survey/SurveyFragmentPresenter.kt
- [ ] app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerFragmentPresenter.kt

### Describe the solution you'd like

It actually seems like the original implementation is incorrect. ``hideSoftInputFromWindow`` expects flag values of ``0`` or a combination of ``HIDE_IMPLICIT_ONLY`` and ``HIDE_NOT_ALWAYS``.

Unfortunately, the documentation is unclear on what '0' indicates, but the other flags seem to indicate cases when the keyboard _shouldn't_ be hidden so it's my interpretation that '0' means 'force hide when possible' (which is what we seem to want in this situation). Thus, the fix seems to be to migrate over to using the '0' constant.

### Describe alternatives you've considered

_No response_

### Additional context

Note that fixes to the files listed above need to be carefully tested to ensure that the keyboard does still hide when expected for each of the scenarios we expect it to hide.

### PR DESCRIPTION
**Pull request by subhajitxyz**
## Explanation
Fixes #5406
    
    Replaced deprecated InputMethodManager.SHOW_FORCED with '0' flag to ensure compatibility with SDK 33.
    Updated the following files:
	    StateFragmentPresenter.kt
	    SurveyFragmentPresenter.kt
	    QuestionPlayerFragmentPresenter.kt
    These changes ensure that the keyboard hides as expected across the app scenarios, maintaining the intended behavior.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by subhajitxyz** [comment]
Now i have signed CLA.

**Comment by Vishwajith-Shettigar** [comment]
Hi @subhajitxyz, thanks for creating a PR. Could you please update your PR according to the essential checklist?

**Comment by subhajitxyz** [comment]
Thank you. I am updating.

**Review by subhajitxyz — COMMENTED** [review]
Checked.

**Comment by Vishwajith-Shettigar** [comment]
@subhajitxyz update title also. Take  #5382  for reference.

**Comment by subhajitxyz** [comment]
> @subhajitxyz update title also. Take #5382 for reference.

I have updated the title as per your reference . Please review and let me know if there are any additional adjustments needed.

**Comment by Vishwajith-Shettigar** [comment]
@subhajitxyz replace **Fixes** with **Fix** in title, and assign @adhiamboperes.

**Comment by subhajitxyz** [comment]
> @subhajitxyz replace **Fixes** with **Fix** in title, and assign @adhiamboperes.

I have updated the title by replacing Fixes with Fix and assigned the issue to @adhiamboperes. Please check and let me know if everything looks good now.

**Comment by adhiamboperes** [comment]
@theMr17, PTAL.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt [hunk 1]
```diff
@@ -438,9 +438,8 @@ class StateFragmentPresenter @Inject constructor(
     val inputManager: InputMethodManager =
       activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
     inputManager.hideSoftInputFromWindow(
-      fragment.view!!.windowToken,
-      @Suppress("DEPRECATION") // TODO(#5406): Use the correct constant value here.
-      InputMethodManager.SHOW_FORCED
+      fragment.requireView().windowToken,
+      0
```
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]:
```suggestion
      fragment.requireView().windowToken,
      0 // Flag value to force hide the keyboard when possible.
```

Adding a comment here and the other places will improve the readability of the code.

**Review by theMr17 — CHANGES_REQUESTED** [review]
Thanks @subhajitxyz! Had one minor comment, PTAL.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt (hunk 1 above)
Comment (subhajitxyz) [reply app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]: added comments in required places. please check it once.

**Review by theMr17 — APPROVED** [review]
Thanks @subhajitxyz, this LGTM!

**Review by adhiamboperes — APPROVED** [review]
LGTM, thanks @subhajitxyz!

Thanks @Vishwajith-Shettigar and @theMr17 as well!

**Comment by subhajitxyz** [comment]
Thank you all for guiding me through my first contribution.

**Pull request merged**